src/middleware.rs

```diff
@@ -139,3 +139,9 @@ impl<'a, State: 'static> Next<'a, State> {
         }
     }
 }
+
+impl<State> Debug for Next<'_, State> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Next").finish()
```

Comment:
Since Middleware has a `name()` now, would it make sense for this to include `self.next_middleware.iter().map(|m| m.name())` somehow? So inspecting Next would make clear what came after the middleware?